tests/test_registry.py

```diff
@@ -39,16 +89,25 @@ def no_type_annotation(x):
         """Return back x"""
         return x
 
-    with pytest.raises(Exception, match="Type annotation must be a JSON serializable type"):
+    with pytest.raises(
+        Exception,
+        match=f"`x` parameter of no_type_annotation must have a JSON-serializable type annotation",
+    ):
         generate_function_schema(no_type_annotation)
 
 
 def test_generate_function_schema_unallowed_type():
-    def unallowed_type(x: set):
+    class NewType:
+        pass
+
+    def unallowed_type(x: NewType):
         '''Return back x'''
         return x
 
-    with pytest.raises(Exception, match="Type annotation must be a JSON serializable type"):
+    with pytest.raises(
+        ValueError,
+        match="Value not declarable with JSON Schema, field: name='x' type=NewType required=True",
```

Comment:
This is what pydantic raises when it comes across a non-JSON-serializable type if `allow_arbitrary_types` is enabled. (If `allow_arbitrary_types` isn't enabled via `FunctionSchemaConfig`, we get a separate less-helpful error.) I was on the fence about floating this error up versus adding some logic back in to catch JSON serialization issues earlier with a custom error message. What do you think @rgbkrk?

Reply:
I think it's a good idea. We can always add it later though.